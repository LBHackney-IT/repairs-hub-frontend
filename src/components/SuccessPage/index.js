import PropTypes from 'prop-types'
import Link from 'next/link'
import WarningText from '../Template/WarningText'

const SuccessPage = ({ ...props }) => {
  return (
    <>
      {props.banner}
      {props.warningText && <WarningText text={props.warningText} />}
      <ul className="lbh-list lbh-!-margin-top-9">
        {props.links.map((link, i) => {
          return (
            <li key={i}>
              <Link href={link.href}>
                <a
                  className="lbh-link"
                  onClick={link.onClick}
                  target={link.target}
                  rel={link.rel}
                >
                  {link.text}
                </a>
              </Link>
            </li>
          )
        })}
      </ul>
    </>
  )
}

SuccessPage.propTypes = {
  links: PropTypes.array.isRequired,
  banner: PropTypes.any,
  warningText: PropTypes.string,
}

export default SuccessPage
