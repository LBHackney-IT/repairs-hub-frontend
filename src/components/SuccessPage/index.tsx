import Link from 'next/link'
import WarningText from '../Template/WarningText'

interface Props {
  links: {
    href: string
    text: string
    rel?: string
    target?: string
    onClick?: () => void
  }[]
  banner: any
  warningText?: string
}

const SuccessPage = (props: Props) => {
  const { links, banner, warningText } = props

  return (
    <>
      {banner}
      {warningText && <WarningText text={warningText} />}
      <ul className="lbh-list lbh-!-margin-top-9">
        {links.map((link, i) => {
          return (
            <li key={i}>
              <Link href={link.href}>
                <a
                  className="lbh-link"
                  onClick={link.onClick}
                  target={link.target}
                  rel={link.rel}
                >
                  {link.text}
                </a>
              </Link>
            </li>
          )
        })}
      </ul>
    </>
  )
}

export default SuccessPage
