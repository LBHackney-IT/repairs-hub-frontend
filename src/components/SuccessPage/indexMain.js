// import { useContext } from 'react'
import PropTypes from 'prop-types'
import Link from 'next/link'
import WarningText from '../Template/WarningText'
// import PageAnnouncement from '../Template/PageAnnouncement'
// import { buildDataFromScheduleAppointment } from '@/utils/hact/jobStatusUpdate/notesForm'
// import { frontEndApiRequest } from '@/utils/frontEndApiClient/requests'
// import UserContext from '../UserContext'

const SuccessPage = ({ ...props }) => {
  const { user } = useContext(UserContext)

  //   const openExternalLinkEventHandler = async () => {
  //     const jobStatusUpdate = buildDataFromScheduleAppointment(
  //       props.workOrderReference.toString(),
  //       `${user.name} opened the DRS Web Booking Manager`
  //     )

  //     await frontEndApiRequest({
  //       method: 'post',
  //       path: `/api/jobStatusUpdate`,
  //       requestData: jobStatusUpdate,
  //     })
  //   }

  return (
    <>
      {props.banner}
      {props.warningText && <WarningText text={props.warningText} />}
      <ul className="lbh-list lbh-!-margin-top-9">
        {props.links.map((link, i) => {
          return (
            <li key={i}>
              <Link href={link.href}>
                <a className="lbh-link" onClick={link.onClick}>
                  {link.text}
                </a>
              </Link>
            </li>
          )
        })}
      </ul>
    </>
  )
}

SuccessPage.propTypes = {
  workOrderReference: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  showDashboardLink: PropTypes.bool,
  showNewWorkOrderLink: PropTypes.bool,
  propertyReference: PropTypes.string,
  shortAddress: PropTypes.string,
  showSearchLink: PropTypes.bool,
  authorisationPendingApproval: PropTypes.bool,
  links: PropTypes.array,
  banner: PropTypes.any,
}

export default SuccessPage
