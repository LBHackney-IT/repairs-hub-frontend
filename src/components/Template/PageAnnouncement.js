import PropTypes from 'prop-types'

export const PageAnnouncement = (props) => (
  <section className="text-align-left lbh-page-announcement">
    <h3 className="lbh-page-announcement__title">{props.title}</h3>
    <div className="lbh-page-announcement__content">
      <p>Reference number</p>
      <strong className="govuk-!-font-size-24">
        {props.workOrderReference}
      </strong>
    </div>
  </section>
)

PageAnnouncement.propTypes = {
  title: PropTypes.string.isRequired,
  workOrderReference: PropTypes.string.isRequired,
}

export default PageAnnouncement
