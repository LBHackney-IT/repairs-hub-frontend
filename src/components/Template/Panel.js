import PropTypes from 'prop-types'

export const Panel = (props) => (
  <div className="govuk-panel govuk-panel--confirmation lbh-panel">
    <h1 className="govuk-panel__title govuk-!-margin-bottom-1">
      {props.title} {props.authorisationText}
    </h1>
    <div className="govuk-panel__body govuk-!-margin-top-0">
      <p>Reference number {props.workOrderReference}</p>

      {props.dateSelected && (
        <p>
          {props.dateSelected} {`${props.slot && props.slot.split(' ')[0]}`}
        </p>
      )}
      {props.comments && (
        <p className="govuk-!-margin-top-1">Comments: {props.comments}</p>
      )}
    </div>
  </div>
)

Panel.propTypes = {
  title: PropTypes.string.isRequired,
  workOrderReference: PropTypes.string.isRequired,
  dateSelected: PropTypes.string,
  slot: PropTypes.string,
  comments: PropTypes.string,
  authorisationText: PropTypes.string,
}

export default Panel
