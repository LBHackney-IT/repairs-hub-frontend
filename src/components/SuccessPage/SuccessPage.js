import { useContext } from 'react'
import PropTypes from 'prop-types'
import Link from 'next/link'
import WarningText from '../Template/WarningText'
import PageAnnouncement from '../Template/PageAnnouncement'
import { buildDataFromScheduleAppointment } from '../../utils/hact/jobStatusUpdate/notesForm'
import { frontEndApiRequest } from '../../utils/frontEndApiClient/requests'
import UserContext from '../UserContext/UserContext'

const SuccessPage = ({ ...props }) => {
  const { user } = useContext(UserContext)

  const openExternalLinkEventHandler = async () => {
    const jobStatusUpdate = buildDataFromScheduleAppointment(
      props.workOrderReference.toString(),
      `${user.name} opened the DRS Web Booking Manager`
    )

    await frontEndApiRequest({
      method: 'post',
      path: `/api/jobStatusUpdate`,
      requestData: jobStatusUpdate,
    })
  }

  return (
    <>
      <PageAnnouncement
        title={props.text}
        workOrderReference={props.workOrderReference}
      />

      {props.authorisationPendingApproval && (
        <WarningText
          text={`Work order ${props.workOrderReference} requires authorisation. Please request authorisation from a manager.`}
        />
      )}

      <ul className="lbh-list lbh-!-margin-top-9">
        {props.externalSchedulerLink && (
          <li>
            Please{' '}
            <Link href={props.externalSchedulerLink}>
              <a
                className="lbh-link"
                target="_blank"
                rel="noopener"
                onClick={openExternalLinkEventHandler}
              >
                <strong>open DRS</strong>
              </a>
            </Link>{' '}
            to book an appointment
          </li>
        )}

        {props.immediateOrEmergencyDloRepairText && (
          <li>
            <p>
              <strong>
                Emergency and immediate DLO repairs are sent directly to the
                Planners. An appointment does not need to be booked.
              </strong>
            </p>
          </li>
        )}

        {props.workOrderReference && (
          <li>
            <Link href={`/work-orders/${props.workOrderReference}`}>
              <a className="lbh-link">
                <strong>View work order</strong>
              </a>
            </Link>
          </li>
        )}

        {props.shortAddress && (
          <li>
            <Link href={`/properties/${props.propertyReference}`}>
              <a className="lbh-link">
                <strong>Back to {props.shortAddress}</strong>
              </a>
            </Link>
          </li>
        )}

        {props.showSearchLink && (
          <li>
            <Link href="/">
              <a className="lbh-link">
                <strong>Start a new search</strong>
              </a>
            </Link>
          </li>
        )}

        {props.showDashboardLink && (
          <li>
            <Link href="/">
              <a className="lbh-link">
                <strong>Back to dashboard</strong>
              </a>
            </Link>
          </li>
        )}

        {props.showNewWorkOrderLink && (
          <li>
            <Link
              href={`/properties/${props.propertyReference}/raise-repair/new`}
            >
              <a className="lbh-link">
                <strong>Raise a new work order</strong>
              </a>
            </Link>
          </li>
        )}
      </ul>
    </>
  )
}

SuccessPage.propTypes = {
  workOrderReference: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  showDashboardLink: PropTypes.bool,
  showNewWorkOrderLink: PropTypes.bool,
  propertyReference: PropTypes.string,
  shortAddress: PropTypes.string,
  showSearchLink: PropTypes.bool,
  authorisationPendingApproval: PropTypes.bool,
}

export default SuccessPage
