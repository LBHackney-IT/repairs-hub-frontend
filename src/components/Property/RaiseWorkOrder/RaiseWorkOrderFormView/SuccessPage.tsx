import {
  createWOLinks,
  LinksWithDRSBooking,
} from '@/root/src/utils/successPageLinks'
import { Property } from '@/root/src/models/propertyTenure'
import { CurrentUser } from '@/root/src/types/variations/types'
import SuccessPage from '@/components/SuccessPage'
import Panel from '../../../Template/Panel'

interface Props {
  immediateOrEmergencyRepairText: boolean
  immediateOrEmergencyDLO: boolean
  authorisationPendingApproval: boolean
  workOrderReference: string
  externallyManagedAppointment: boolean
  property: Property
  currentUser: CurrentUser
  externalAppointmentManagementUrl: string
}

const RaiseWorkOrderSuccessPage = (props: Props) => {
  const {
    immediateOrEmergencyRepairText,
    immediateOrEmergencyDLO,
    authorisationPendingApproval,
    workOrderReference,
    externallyManagedAppointment,
    property,
    currentUser,
    externalAppointmentManagementUrl,
  } = props

  const warningTextToShow = () => {
    if (immediateOrEmergencyRepairText) {
      if (immediateOrEmergencyDLO) {
        return 'Emergency and immediate DLO repairs are sent directly to the planners. An appointment does not need to be booked'
      } else
        return 'Emergency and immediate repairs must be booked immediately. Please call the external contractor'
    } else if (authorisationPendingApproval) {
      return 'Please request authorisation from a manager'
    } else {
      return ''
    }
  }

  return (
    <SuccessPage
      banner={
        <Panel
          title="Work order created"
          authorisationText={
            authorisationPendingApproval && 'but requires authorisation'
          }
          workOrderReference={workOrderReference}
        />
      }
      warningText={warningTextToShow()}
      links={
        externallyManagedAppointment
          ? LinksWithDRSBooking(
              workOrderReference,
              property,
              externalAppointmentManagementUrl,
              currentUser.name
            )
          : createWOLinks(workOrderReference, property)
      }
    />
  )
}

export default RaiseWorkOrderSuccessPage
