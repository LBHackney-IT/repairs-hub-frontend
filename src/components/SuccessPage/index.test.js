import { render } from '@testing-library/react'
import SuccessPage from './index'
import Panel from '@/components/Template/Panel'
import PageAnnouncement from '@/components/Template/PageAnnouncement'
import {
  createWOLinks,
  LinksWithDRSBooking,
  cancelWorkOrderLinks,
  generalLinks,
  updateWorkOrderLinks,
  rejectLinks,
  authorisationApprovedLinks,
} from '@/utils/successPageLinks'

describe('SuccessPage component', () => {
  const props = {
    workOrder: {
      workOrderReference: '10000012',
    },
    property: {
      propertyReference: '12345678',
      address: {
        shortAddress: '16 Pitcairn House  St Thomass Square',
        addressLine: '16 Pitcairn House',
      },
    },
  }

  describe('Raising a repair', () => {
    describe('creates WO with RH appointment', () => {
      it('should render success message with appointment date and related links', () => {
        const { asFragment } = render(
          <SuccessPage
            banner={
              <Panel
                title="Work order created"
                workOrderReference={props.workOrder.workOrderReference}
                dateSelected={'Thursday, 11 March'}
                slot={'AM'}
                comments={'10 am works for me'}
              />
            }
            links={createWOLinks(
              props.workOrder.workOrderReference,
              props.property
            )}
          />
        )
        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('creates WO with an external scheduler URL', () => {
      it('should render text with a link to schedule externally', () => {
        const externalSchedulerUrl = 'https://drs.example.com'
        const userName = 'Hackney User'

        const { asFragment } = render(
          <SuccessPage
            banner={
              <Panel
                title="Work order created"
                workOrderReference={props.workOrder.workOrderReference}
              />
            }
            links={LinksWithDRSBooking(
              props.workOrder.workOrderReference,
              props.property,
              externalSchedulerUrl,
              userName
            )}
          />
        )
        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('creates Emergency/Immediate DLO WO', () => {
      it('should render a success page with an Emergency/Immediate DLO warning text', () => {
        const { asFragment } = render(
          <SuccessPage
            banner={
              <Panel
                title="Work order created"
                workOrderReference={props.workOrder.workOrderReference}
              />
            }
            warningText="Emergency and immediate DLO repairs are sent directly to the planners. An appointment does not need to be booked"
            links={createWOLinks(
              props.workOrder.workOrderReference,
              props.property
            )}
          />
        )
        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('creates Emergency/Immediate WO for external contractor', () => {
      it('should render a success page with an Emergency/Immediate warning text', () => {
        const { asFragment } = render(
          <SuccessPage
            banner={
              <Panel
                title="Work order created"
                workOrderReference={props.workOrder.workOrderReference}
              />
            }
            warningText="Emergency and immediate repairs must be booked immediately. Please call the external contractor"
            links={createWOLinks(
              props.workOrder.workOrderReference,
              props.property
            )}
          />
        )
        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('High cost (over raise limit) authorisation', () => {
      it('should render a success screen with a warning message', () => {
        const { asFragment } = render(
          <SuccessPage
            banner={
              <Panel
                title="Work order created but requires authorisation"
                workOrderReference={props.workOrder.workOrderReference}
              />
            }
            warningText="Please request authorisation from a manager"
            links={createWOLinks(
              props.workOrder.workOrderReference,
              props.property
            )}
          />
        )
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
  describe('WO cancelled', () => {
    it('should render a success screen with a link to raise a new wo', () => {
      const { asFragment } = render(
        <SuccessPage
          banner={
            <Panel
              title="Work order cancelled"
              workOrderReference={props.workOrder.workOrderReference}
            />
          }
          links={cancelWorkOrderLinks(
            props.workOrder.workOrderReference,
            props.propertyReference,
            props.property.address.shortAddress
          )}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('WO closed', () => {
    it('should render a success screen', () => {
      const { asFragment } = render(
        <SuccessPage
          banner={
            <Panel
              title="Work order closed"
              workOrderReference={props.workOrder.workOrderReference}
            />
          }
          links={generalLinks(props.workOrder.workOrderReference)}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('WO update', () => {
    it('should render a success screen', () => {
      const { asFragment } = render(
        <SuccessPage
          banner={
            <PageAnnouncement
              title="Work order updated"
              workOrderReference={props.workOrder.workOrderReference}
            />
          }
          links={updateWorkOrderLinks(props.workOrder.workOrderReference)}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('Variation requires authorisation', () => {
    it('should render a success screen with warning text', () => {
      const { asFragment } = render(
        <SuccessPage
          banner={
            <PageAnnouncement
              title="Variation requires authorisation"
              workOrderReference={props.workOrder.workOrderReference}
            />
          }
          links={generalLinks(props.workOrder.workOrderReference)}
          showWarningText={true}
          warningText="Please request authorisation from a manager"
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('Variation approved', () => {
    it('should render a success screen', () => {
      const { asFragment } = render(
        <SuccessPage
          banner={
            <PageAnnouncement
              title="Variation request approved"
              workOrderReference={props.workOrder.workOrderReference}
            />
          }
          links={generalLinks(props.workOrder.workOrderReference)}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('Variation rejected', () => {
    it('should render a success screen with link to raise a new wo', () => {
      const { asFragment } = render(
        <SuccessPage
          banner={
            <PageAnnouncement
              title="Variation request rejected"
              workOrderReference={props.workOrder.workOrderReference}
            />
          }
          links={rejectLinks(
            props.workOrder.workOrderReference,
            props.property.propertyReference,
            props.property.address.shortAddress
          )}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('Authorisation approved', () => {
    it('should render a success screen with link to book an appointment', () => {
      const { asFragment } = render(
        <SuccessPage
          banner={
            <PageAnnouncement
              title="Authorisation request approved"
              workOrderReference={props.workOrder.workOrderReference}
            />
          }
          links={authorisationApprovedLinks(props.workOrder.workOrderReference)}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('Authorisation rejected', () => {
    it('should render a success screen with link to raise a new wo', () => {
      const { asFragment } = render(
        <SuccessPage
          banner={
            <Panel
              title="Work order cancelled, authorisation request rejected"
              workOrderReference={props.workOrder.workOrderReference}
            />
          }
          links={cancelWorkOrderLinks(
            props.workOrder.workOrderReference,
            props.propertyReference,
            props.property.address.shortAddress
          )}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
