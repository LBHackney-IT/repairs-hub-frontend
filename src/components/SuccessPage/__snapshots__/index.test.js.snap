// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessPage component Authorisation approved should render a success screen with link to book an appointment 1`] = `
<DocumentFragment>
  <section
    class="text-align-left lbh-page-announcement"
  >
    <h3
      class="lbh-page-announcement__title"
    >
      Authorisation request approved
    </h3>
    <div
      class="lbh-page-announcement__content"
    >
      <p>
        Reference number
      </p>
      <strong
        class="govuk-!-font-size-24"
      >
        10000012
      </strong>
    </div>
  </section>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012/appointment/new"
      >
        Book an appointment on DRS
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Manage work orders
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Authorisation rejected should render a success screen with link to raise a new wo 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel"
  >
    <h1
      class="govuk-panel__title govuk-!-margin-bottom-1"
    >
      Work order cancelled, authorisation request rejected 
    </h1>
    <div
      class="govuk-panel__body govuk-!-margin-top-0"
    >
      <p>
        Reference number 10000012
      </p>
    </div>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/properties/undefined/raise-repair/new"
      >
        Raise a new work order for 16 Pitcairn House  St Thomass Square
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Start a new search
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Raising a repair High cost (over raise limit) authorisation should render a success screen with a warning message 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel"
  >
    <h1
      class="govuk-panel__title govuk-!-margin-bottom-1"
    >
      Work order created but requires authorisation 
    </h1>
    <div
      class="govuk-panel__body govuk-!-margin-top-0"
    >
      <p>
        Reference number 10000012
      </p>
    </div>
  </div>
  <div
    class="govuk-warning-text lbh-warning-text"
  >
    <span
      aria-hidden="true"
      class="govuk-warning-text__icon"
    >
      !
    </span>
    <strong
      class="govuk-warning-text__text"
    >
      <span
        class="govuk-warning-text__assistive"
      >
        Warning
      </span>
      Please request authorisation from a manager.
    </strong>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/properties/12345678"
      >
        Back to 16 Pitcairn House
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Start a new search
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Raising a repair creates Emergency/Immediate DLO WO should render a success page with an Emergency/Immediate DLO warning text 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel"
  >
    <h1
      class="govuk-panel__title govuk-!-margin-bottom-1"
    >
      Work order created 
    </h1>
    <div
      class="govuk-panel__body govuk-!-margin-top-0"
    >
      <p>
        Reference number 10000012
      </p>
    </div>
  </div>
  <div
    class="govuk-warning-text lbh-warning-text"
  >
    <span
      aria-hidden="true"
      class="govuk-warning-text__icon"
    >
      !
    </span>
    <strong
      class="govuk-warning-text__text"
    >
      <span
        class="govuk-warning-text__assistive"
      >
        Warning
      </span>
      Emergency and immediate DLO repairs are sent directly to the planners. An appointment does not need to be booked.
    </strong>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/properties/12345678"
      >
        Back to 16 Pitcairn House
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Start a new search
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Raising a repair creates Emergency/Immediate WO for external contractor should render a success page with an Emergency/Immediate warning text 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel"
  >
    <h1
      class="govuk-panel__title govuk-!-margin-bottom-1"
    >
      Work order created 
    </h1>
    <div
      class="govuk-panel__body govuk-!-margin-top-0"
    >
      <p>
        Reference number 10000012
      </p>
    </div>
  </div>
  <div
    class="govuk-warning-text lbh-warning-text"
  >
    <span
      aria-hidden="true"
      class="govuk-warning-text__icon"
    >
      !
    </span>
    <strong
      class="govuk-warning-text__text"
    >
      <span
        class="govuk-warning-text__assistive"
      >
        Warning
      </span>
      Emergency and immediate repairs must be booked immediately. Please call the external contractor.
    </strong>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/properties/12345678"
      >
        Back to 16 Pitcairn House
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Start a new search
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Raising a repair creates WO with RH appointment should render success message with appointment date and related links 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel"
  >
    <h1
      class="govuk-panel__title govuk-!-margin-bottom-1"
    >
      Work order created 
    </h1>
    <div
      class="govuk-panel__body govuk-!-margin-top-0"
    >
      <p>
        Reference number 10000012
      </p>
      <p>
        Thursday, 11 March AM
      </p>
      <p
        class="govuk-!-margin-top-1"
      >
        Comments: 10 am works for me
      </p>
    </div>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/properties/12345678"
      >
        Back to 16 Pitcairn House
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Start a new search
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Raising a repair creates WO with an external scheduler URL should render text with a link to schedule externally 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel"
  >
    <h1
      class="govuk-panel__title govuk-!-margin-bottom-1"
    >
      Work order created 
    </h1>
    <div
      class="govuk-panel__body govuk-!-margin-top-0"
    >
      <p>
        Reference number 10000012
      </p>
    </div>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="https://drs.example.com"
        rel="noopener"
        target="_blank"
      >
        Book an appointment on DRS
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/properties/12345678"
      >
        Back to 16 Pitcairn House
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Start a new search
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Variation approved should render a success screen 1`] = `
<DocumentFragment>
  <section
    class="text-align-left lbh-page-announcement"
  >
    <h3
      class="lbh-page-announcement__title"
    >
      Variation request approved
    </h3>
    <div
      class="lbh-page-announcement__content"
    >
      <p>
        Reference number
      </p>
      <strong
        class="govuk-!-font-size-24"
      >
        10000012
      </strong>
    </div>
  </section>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Manage work orders
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Variation rejected should render a success screen with link to raise a new wo 1`] = `
<DocumentFragment>
  <section
    class="text-align-left lbh-page-announcement"
  >
    <h3
      class="lbh-page-announcement__title"
    >
      Variation request rejected
    </h3>
    <div
      class="lbh-page-announcement__content"
    >
      <p>
        Reference number
      </p>
      <strong
        class="govuk-!-font-size-24"
      >
        10000012
      </strong>
    </div>
  </section>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/properties/12345678/raise-repair/new"
      >
        Raise a new work order for 16 Pitcairn House  St Thomass Square
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Manage work orders
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component Variation requires authorisation should render a success screen with warning text 1`] = `
<DocumentFragment>
  <section
    class="text-align-left lbh-page-announcement"
  >
    <h3
      class="lbh-page-announcement__title"
    >
      Variation requires authorisation
    </h3>
    <div
      class="lbh-page-announcement__content"
    >
      <p>
        Reference number
      </p>
      <strong
        class="govuk-!-font-size-24"
      >
        10000012
      </strong>
    </div>
  </section>
  <div
    class="govuk-warning-text lbh-warning-text"
  >
    <span
      aria-hidden="true"
      class="govuk-warning-text__icon"
    >
      !
    </span>
    <strong
      class="govuk-warning-text__text"
    >
      <span
        class="govuk-warning-text__assistive"
      >
        Warning
      </span>
      Please request authorisation from a manager.
    </strong>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Manage work orders
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component WO cancelled should render a success screen with a link to raise a new wo 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel"
  >
    <h1
      class="govuk-panel__title govuk-!-margin-bottom-1"
    >
      Work order cancelled 
    </h1>
    <div
      class="govuk-panel__body govuk-!-margin-top-0"
    >
      <p>
        Reference number 10000012
      </p>
    </div>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/properties/undefined/raise-repair/new"
      >
        Raise a new work order for 16 Pitcairn House  St Thomass Square
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Start a new search
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component WO closed should render a success screen 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel"
  >
    <h1
      class="govuk-panel__title govuk-!-margin-bottom-1"
    >
      Work order closed 
    </h1>
    <div
      class="govuk-panel__body govuk-!-margin-top-0"
    >
      <p>
        Reference number 10000012
      </p>
    </div>
  </div>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Manage work orders
      </a>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SuccessPage component WO update should render a success screen 1`] = `
<DocumentFragment>
  <section
    class="text-align-left lbh-page-announcement"
  >
    <h3
      class="lbh-page-announcement__title"
    >
      Work order updated
    </h3>
    <div
      class="lbh-page-announcement__content"
    >
      <p>
        Reference number
      </p>
      <strong
        class="govuk-!-font-size-24"
      >
        10000012
      </strong>
    </div>
  </section>
  <ul
    class="lbh-list lbh-!-margin-top-9"
  >
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012"
      >
        View work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/work-orders/10000012/close"
      >
        Close work order
      </a>
    </li>
    <li>
      <a
        class="lbh-link"
        href="/"
      >
        Manage work orders
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
